tools/sphere.c

```diff
@@ -80,3 +78,31 @@ int do_touch( string str ) {
 int do_gaze( string str ) {
 
 }
+
+/**
+ *  Function finds and returns the sloop shadow based on the
+ *  string passed.
+ *  @param {String} sloop_type : valid values either "slooper" or "slooped"
+ *  @param {Object} ob :  player object to test for shadows
+ *  @return {Object} : returns the shadow object
+ */
+
+string find_sloop_shadow( string sloop_type, object ob ) {
+    if ( sloop_type != "slooper" || sloop_type != "slooped" ) {
+        return 0;
+    }
+    if (! ob ) {
+        return 0;
+    }
+    object *list_shadows = shadow_list( ob );
+    int i;
+    for ( i = 0; i <= sizeof( list_shadows ); i += 1  ) {
+        string shadow_name = program_name( list_shadows[i] );
+        if ( shadow_name == SLOOPER_SHAD ) {
+            return shadow_name;
+        }
+        if ( shadow_name == SLOOPEE_SHAD ) {
+            return shadow_name;
+        }
+    }
+}
```

Comment:
I like putting a `return 0;` here at the end of your function, just to be explicit in the fact that you return 0 if you don't find any matching shadow. Functions return 0 by default if it gets to the end without returning anything else, so it's 100% not necessary. It's really just a style thing.

Note that if the end of the function is unreachable (for instance, you have an if/else block or loop body that is certain to always return something), then I wouldn't bother. In that situation, you're still explicitly returning even if it's not at the end of the function body, so the default would never kick in.